cdap-app-templates/cdap-etl/cdap-etl-core/src/main/java/co/cask/cdap/templates/etl/batch/ETLBatchTemplate.java

```diff
@@ -0,0 +1,138 @@
+
+  public ETLBatchTemplate() throws Exception {
+    nameToClass = HashBasedTable.create();
+    //TODO: Add classes from Lib here to be available for use in the ETL Adapter. Remove this when
+    //plugins management is completed.
+    initTable(Lists.<Class>newArrayList(KVTableSource.class, KVTableSink.class, IdentityTransform.class));
+  }
+
+  private void initTable(List<Class> classList) throws Exception {
+    for (Class klass : classList) {
+      DefaultStageConfigurer configurer = new DefaultStageConfigurer(klass);
+      if (BatchSource.class.isAssignableFrom(klass)) {
+        BatchSource source = (BatchSource) klass.newInstance();
+        source.configure(configurer);
+        nameToClass.put("source", configurer.createSpecification().getName(),
+                        configurer.createSpecification().getClassName());
+      } else if (BatchSink.class.isAssignableFrom(klass)) {
+        BatchSink sink = (BatchSink) klass.newInstance();
+        sink.configure(configurer);
+        nameToClass.put("sink", configurer.createSpecification().getName(),
+                        configurer.createSpecification().getClassName());
+      } else {
+        Preconditions.checkArgument(Transform.class.isAssignableFrom(klass));
+        Transform transform = (Transform) klass.newInstance();
+        transform.configure(configurer);
+        nameToClass.put("transform", configurer.createSpecification().getName(),
+                        configurer.createSpecification().getClassName());
+      }
+    }
+  }
+
+  @Override
+  public void configureAdapter(String adapterName, ETLBatchConfig etlConfig, AdapterConfigurer configurer)
+    throws Exception {
+    // Get cronEntry string for ETL Batch Adapter
+    String cronEntry = etlConfig.getSchedule();
+
+    ETLStage source = etlConfig.getSource();
+    ETLStage sink = etlConfig.getSink();
+    List<ETLStage> transform = etlConfig.getTransforms();
+
+    configureSource(source, configurer);
+    configureSink(sink, configurer);
+    configureTransform(transform, configurer);
+
+    //TODO: Validate if source, transforms, sink can be tied together
+
+    configurer.addRuntimeArgument(Constants.ADAPTER_NAME, adapterName);
+    configurer.setSchedule(new TimeSchedule(String.format("etl.batch.adapter.%s.schedule", adapterName),
+                                            String.format("Schedule for %s Adapter", adapterName),
+                                            cronEntry));
+  }
+
+  private void configureSource(ETLStage source, AdapterConfigurer configurer) throws Exception {
+    String sourceName = source.getName();
+    String className = nameToClass.get("source", sourceName);
+    BatchSource batchSource = (BatchSource) Class.forName(className).newInstance();
+    DefaultStageConfigurer stageConfigurer = new DefaultStageConfigurer(batchSource.getClass());
+    StageSpecification specification = stageConfigurer.createSpecification();
```

Comment:
not sure I understand what is going on here.  Isn't this where we want to be calling configure on the source?  And shouldn't it be passing the AdapterConfigurer to them?